c2rust-analyze/src/main.rs

```diff
@@ -321,6 +321,46 @@ where
     }
 }
 
+fn mark_foreign_fixed<'tcx>(
+    gacx: &mut GlobalAnalysisCtxt<'tcx>,
+    gasn: &mut GlobalAssignment,
+    tcx: TyCtxt<'tcx>,
+) {
+    // FIX the inputs and outputs of function declarations in extern blocks
+    for did in gacx.foreign_mentioned_tys.clone() {
+        if let DefKind::Fn = tcx.def_kind(did) {
+            let sig = tcx.erase_late_bound_regions(tcx.fn_sig(did));
+            let inputs = sig
+                .inputs()
+                .iter()
+                .map(|&ty| gacx.assign_pointer_ids_with_info(ty, PointerInfo::ANNOTATED))
```

Comment:
Is it right to reuse `ANNOTATED` for what when we mean `FIXED`?

Reply:
Or is this `ANNOTATED` merely because it's from a `fn` signature, and those must be annotated?

Reply:
`ANNOTATED` is correct because there's an explicit type annotation in the source code corresponding to these `Ty`s.  Normal `fn`s are similarly marked `ANNOTATED`.